internal/gw/client_publish_test.go

```diff
@@ -89,7 +89,7 @@ func TestClientGetPublish(t *testing.T) {
 	gw.publishes["some-id"] = &fakePublish{id: "some-id"}
 
 	// It should be able to get a publish
-	p := clientIface.GetPublish("some-id")
+	p, _ := clientIface.GetPublish(ctx, "some-id")
```

Comment:
Please verify also that `err` is nil.  I think right now if you try to verify that, you'll find there's a bug in the test.